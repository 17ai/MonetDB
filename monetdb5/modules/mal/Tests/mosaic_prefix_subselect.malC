b:= bat.new(:oid,:int);
bat.append(b,50);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,51);
bat.append(b,60);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,67);
bat.append(b,255);
bat.append(b,256);
bat.append(b,256);
bat.append(b,256);
bat.append(b,256);
bat.append(b,256);
bat.append(b,255);
bat.append(b,255);
bat.append(b,255);
bat.append(b,255);
bat.append(b,255);

x:= mosaic.compress(b,"prefix");

s:= algebra.subselect(b,60,68,false,false,false);
io.print(s);
xs:= mosaic.subselect(x,60,68,false,false,false);
io.print(xs);

s:= algebra.subselect(b,51,62,false,false,false);
io.print(s);
xs:= mosaic.subselect(x,51,62,false,false,false);
io.print(xs);

s:= algebra.subselect(b,60,68,true,true,false);
io.print(s);
xs:= mosaic.subselect(x,60,68,true,true,false);
io.print(xs);

s:= algebra.subselect(b,51,62,true,true,false);
io.print(s);
xs:= mosaic.subselect(x,51,62,true,true,false);
io.print(xs);
