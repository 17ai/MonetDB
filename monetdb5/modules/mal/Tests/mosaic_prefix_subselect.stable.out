stdout of test 'mosaic_prefix_subselect` in directory 'monetdb5/modules/mal` itself:


# 12:33:31 >  
# 12:33:31 >  "mserver5" "--debug=10" "--set" "gdk_nr_threads=0" "--set" "mapi_open=true" "--set" "mapi_port=33381" "--set" "mapi_usock=/var/tmp/mtest-25823/.s.monetdb.33381" "--set" "monet_prompt=" "--forcemito" "--set" "mal_listing=2" "--dbpath=/export/scratch1/mk/mosaic//Linux/var/MonetDB/mTests_monetdb5_modules_mal" "mosaic_prefix_subselect.mal"
# 12:33:31 >  

# MonetDB 5 server v11.20.0
# This is an unreleased version
# Serving database 'mTests_monetdb5_modules_mal', using 8 threads
# Compiled for x86_64-unknown-linux-gnu/64bit with 64bit OIDs and 128bit integers dynamically linked
# Found 15.590 GiB available main-memory.
# Copyright (c) 1993-July 2008 CWI.
# Copyright (c) August 2008-2014 MonetDB B.V., all rights reserved
# Visit http://www.monetdb.org/ for further information
# Listening for connection requests on mapi:monetdb://vienna.ins.cwi.nl:33381/
# Listening for UNIX domain connection requests on mapi:monetdb:///var/tmp/mtest-25823/.s.monetdb.33381
# MonetDB/GIS module loaded
# MonetDB/SQL module loaded
function user.main():void;
    b := bat.new(:oid,:int);
    bat.append(b,50);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,51);
    bat.append(b,60);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,67);
    bat.append(b,255);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,256);
    bat.append(b,255);
    bat.append(b,255);
    bat.append(b,255);
    bat.append(b,255);
    bat.append(b,255);
    x := mosaic.compress(b,"prefix");
    s := algebra.subselect(b,60,68,false,false,false);
    io.print(s);
    xs := mosaic.subselect(x,60,68,false,false,false);
    io.print(xs);
    s := algebra.subselect(b,51,62,false,false,false);
    io.print(s);
    xs := mosaic.subselect(x,51,62,false,false,false);
    io.print(xs);
    s := algebra.subselect(b,60,68,true,true,false);
    io.print(s);
    xs := mosaic.subselect(x,60,68,true,true,false);
    io.print(xs);
    s := algebra.subselect(b,51,62,true,true,false);
    io.print(s);
    xs := mosaic.subselect(x,51,62,true,true,false);
    io.print(xs);
end main;
#--------------------------#
# h	t  # name
# void	void  # type
#--------------------------#
[ 0@0,	8@0  ]
[ 1@0,	9@0  ]
[ 2@0,	10@0  ]
[ 3@0,	11@0  ]
[ 4@0,	12@0  ]
[ 5@0,	13@0  ]
[ 6@0,	14@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	8@0  ]
[ 1@0,	9@0  ]
[ 2@0,	10@0  ]
[ 3@0,	11@0  ]
[ 4@0,	12@0  ]
[ 5@0,	13@0  ]
[ 6@0,	14@0  ]
#--------------------------#
# h	t  # name
# void	void  # type
#--------------------------#
[ 0@0,	7@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	7@0  ]
#--------------------------#
# h	t  # name
# void	void  # type
#--------------------------#
[ 0@0,	7@0  ]
[ 1@0,	8@0  ]
[ 2@0,	9@0  ]
[ 3@0,	10@0  ]
[ 4@0,	11@0  ]
[ 5@0,	12@0  ]
[ 6@0,	13@0  ]
[ 7@0,	14@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	7@0  ]
[ 1@0,	8@0  ]
[ 2@0,	9@0  ]
[ 3@0,	10@0  ]
[ 4@0,	11@0  ]
[ 5@0,	12@0  ]
[ 6@0,	13@0  ]
[ 7@0,	14@0  ]
#--------------------------#
# h	t  # name
# void	void  # type
#--------------------------#
[ 0@0,	1@0  ]
[ 1@0,	2@0  ]
[ 2@0,	3@0  ]
[ 3@0,	4@0  ]
[ 4@0,	5@0  ]
[ 5@0,	6@0  ]
[ 6@0,	7@0  ]
#--------------------------#
# h	t  # name
# void	oid  # type
#--------------------------#
[ 0@0,	1@0  ]
[ 1@0,	2@0  ]
[ 2@0,	3@0  ]
[ 3@0,	4@0  ]
[ 4@0,	5@0  ]
[ 5@0,	6@0  ]
[ 6@0,	7@0  ]

# 12:33:32 >  
# 12:33:32 >  "Done."
# 12:33:32 >  

